src/openms/source/FORMAT/XMLFile.cpp

```diff
@@ -152,21 +154,38 @@ namespace OpenMS
       try
       {
         parser->parse(*source);
-        delete(parser);
+        delete parser;
         delete source;
       }
       catch (const xercesc::XMLException & toCatch)
       {
-        throw Exception::ParseError(__FILE__, __LINE__, __PRETTY_FUNCTION__, "", String("XMLException: ") + StringManager().convert(toCatch.getMessage()));
+        delete parser;
```

Comment:
opting for boost::shared_ptr here since ideally you would have
```delete parser;
parser = 0;
...```

do avoid double deletion etc... we can save a lot of code just using shared_ptr.